dsf-gdb/org.eclipse.cdt.dsf.gdb/src/org/eclipse/cdt/dsf/gdb/launching/FinalLaunchSequence.java

```diff
@@ -133,11 +134,9 @@ protected String[] getExecutionOrder(String group) {
 					//
 					// "stepSetSourceLookupPath",   //$NON-NLS-1$
 
-					// For remote-attach launch only
```

Comment:
I'm just thinking out loud now; Would it be safer to continue to call stepRemoteConnection until it is to be purged from the API?
With this change, it will be a "silent" error and it will be harder to find the reason why some other code that might extend this class stopped working. Sure - one should read the N&N, but in reality, this kind of change can easily be missed.

Reply:
Yes, this is what I recommended in https://github.com/eclipse-cdt/cdt/pull/336#pullrequestreview-1371709721 - it isn't just for the call, but code like this too: https://github.com/eclipse-embed-cdt/eclipse-plugins/blob/7115cc9957b44cabb495580f7a0e7452ce8647bf/plugins/org.eclipse.embedcdt.debug.gdbjtag.core/src/org/eclipse/embedcdt/debug/gdbjtag/core/dsf/GnuMcuFinalLaunchSequence.java#L64

Reply:
the more I think about it, I get confused that why should we convolute our code for this case.. the current solution seems more inline with what we do for non-attach remote launches..

however, I have worked on your suggestion and kept the logic inside the two methods with some comments for future when removing the APIs.. but I personally don't like this approach :stuck_out_tongue:

Reply:
Sorry @umairsair - I think there was some confusion as to what the request in this comment was. Your previous change removed the call to stepRemoteConnection, so I think all that was missing was this part of my comment:

```diff
--- a/dsf-gdb/org.eclipse.cdt.dsf.gdb/src/org/eclipse/cdt/dsf/gdb/launching/FinalLaunchSequence.java
+++ b/dsf-gdb/org.eclipse.cdt.dsf.gdb/src/org/eclipse/cdt/dsf/gdb/launching/FinalLaunchSequence.java
@@ -134,6 +134,7 @@ public class FinalLaunchSequence extends ReflectionSequence {
                                        //
                                        // "stepSetSourceLookupPath",   //$NON-NLS-1$
 
+                                       "stepRemoteConnection", //$NON-NLS-1$
                                        // For all launches except attach ones
                                        "stepNewProcess", //$NON-NLS-1$
                                        // For all attach launch only
```


For API reasons the way you have done it now means that we won't break anything which is great and goes beyond what I was hoping you would do because the more I look around the more references to this code in extenders.

But it does leave the code more convoluted.

Also, the stepAttachRemoteToDebugger should still be in this list. The current version is a bit of a mishmash between both cases. I will provide an updated commit that I hope addresses this so we can get it merged in.